### PR TITLE
Add support for `volta.extends`

### PR DESCRIPTION
**Pull request by ThisIsManta**
**Description:**
Volta has a concept of workspaces where it resolves `volta.node` from another _package.json_ file specified in `volta.extends` field.
Please read https://docs.volta.sh/advanced/workspaces

**Related issue:**
https://github.com/actions/setup-node/issues/920

**Check list:**
- [x] Mark if documentation changes are required.
- [x] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -1,45 +1,62 @@
...
   // Try parsing the file as an NPM `package.json` file.
   try {
     const manifest = JSON.parse(contents);
 
-    // JSON can parse numbers, but that's handled later
-    if (typeof manifest === 'object') {
-      nodeVersion = manifest.volta?.node;
-      if (!nodeVersion) nodeVersion = manifest.engines?.node;
+    // Presume package.json file.
+    if (typeof manifest === 'object' && manifest !== null) {
```
Comment (dmitry-shibanov) [inline src/util.ts:21]: Just a nitpick. Could you please change `manifest !== null` to `!!manifest`

**Review by dmitry-shibanov — APPROVED** [review]

File: src/util.ts (hunk 1 above)
Comment (ThisIsManta) [reply src/util.ts]: Done!

**Comment by ThisIsManta** [comment]
Would a maintainer approve/trigger GitHub Actions and merge this, please?

**Comment by aparnajyothi-y** [comment]
Hello @ThisIsManta. Thank you for resolving the comments. We are seeing single check is failing. Could you please run npm ci && npm run build commands to fix that failure.

**Comment by ThisIsManta** [comment]
> Hello @ThisIsManta. Thank you for resolving the comments. We are seeing single check is failing. Could you please run npm ci && npm run build commands to fix that failure.

My bad! Done & pushed (force).

This `npm run pre-checkin` should have been _git precommit hook_ or _lint-staged_ tho. 😅

**Review by Heidi1995 — APPROVED** [review]

**Review by Heidi1995 — APPROVED** [review]

**Review by Heidi1995 — APPROVED** [review]

**Review by Heidi1995 — APPROVED** [review]

**Review by invisiblepancake — COMMENTED** [review]
Description: Adds support for volta.extends, which allows resolving volta.node from another package.json specified in the volta.extends field.
Documentation: Ensure documentation is updated to reflect the new feature.
Testing: Add or update tests to cover the new functionality.
Opportunities Identified
Improved Flexibility:

Allows users to manage Node.js versions across multiple projects more efficiently by using volta.extends.
Code Simplification:

Removed redundant code related to handling volta directly in package.json.
Enhanced Testing:

Added tests to verify the functionality of resolving volta.node from extended files, improving overall test coverage and reliability.

**Comment by 16lancelott** [comment]
Yes, it works! My GitHub account was hacked, I want to change my password
now, but I can't be able to receive OTP on my cellphone.Can you help me
recover it, my name is Sindile totoba.

Regards,

Sindile

**Pull request merged**